src/Http/Http.Abstractions/src/IHttpContextAccessor.cs

```diff
@@ -3,8 +3,14 @@
 
 namespace Microsoft.AspNetCore.Http
 {
+    /// <summary>
+    /// Accessor to the <see cref="HttpContext"/>
+    /// </summary>
```

Comment:
We should have a blurb about why `IHttpContextAccessor` use isn't recommended unless absolutely necessary.

```suggestion
    /// </summary>
    /// <remarks>
    /// This interface should be used with caution. It relies on <see cref="System.Threading.AsyncLocal{T}" /> which can have a negative performance impact on async calls.
    /// It also creates a dependency on "ambient state" which can make testing more difficult.
    /// </remarks>
```